std/machines/memory16.asm

```diff
@@ -0,0 +1,76 @@
+use std::array;
+use std::machines::range::Byte2;
+
+// A read/write memory, similar to that of Polygon:
+// https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/mem.pil
+machine Memory16(byte2: Byte2) with
+    latch: LATCH,
+    operation_id: m_is_write,
+    call_selectors: selectors,
+{
+    // lower bound degree is 65536
+
+    operation mload<0> m_addr, m_step -> m_value1, m_value2;
+    operation mstore<1> m_addr, m_step, m_value1, m_value2 ->;
+
+    let LATCH = 1;
+
+    // =============== read-write memory =======================
+    // Read-write memory. Columns are sorted by addr and
+    // then by step. change is 1 if and only if addr changes
+    // in the next row.
+    // Note that these column names are used by witgen to detect
+    // this machine...
+    col witness m_addr;
+    col witness m_step;
+    col witness m_change;
+    col witness m_value1, m_value2;
+
+    link => byte2.check(m_value1);
+    link => byte2.check(m_value2);
```

Comment:
I guess it's good to have, but strictly speaking not needed for the machine to work properly, right?

Anyway, I think it makes sense to add a comment that we expect the two values to be 16-Bit limbs of a 32-Bit machine word.